### PR TITLE
Refactor CD Pipeline | Split Server from Client

### PR DESCRIPTION
**Pull request by gbmyt**
Split the deploy configuration into 2 separate files: 

> Client now deploys to it's own EC2 while the server is deployed in a separate Github Action to it's own EC2.
      
> The backend now uses `env-cmd` npm package to run the server using our environment variables without us having to commit sensitive data. **_Please re-run npm install once merged to main_**

> I may still need to confirm the 2 EC2 instances can communicate with each other, but that's an AWS issue, just noting here in case any issues pulling data into the frontend from the BE server.
       
> You can confirm the deployed server is running by using POSTMAN to get the following endpoint:

      `GET ec2-3-83-24-209.compute-1.amazonaws.com:8080/quiz/1/start`

> Slightly unrelated but I organized all the route imports in `server/index.js` under one routes comment and made sure the 'running on port xyz' log is at the very bottom of the file. This was aesthetic and shouldn't impact functionality in any way but lmk if you find otherwise.

Please review, thanks!

-----------
UPDATE: 
- Removed obsolete `env-cmd` npm package in favor of using Github Actions to create and copy the dotenv files into our server container on the fly. 
- I also updated the cors header to use the deployed API_URI (uncomment local testing string when developing locally).
- Tested server pipeline / DB_URL connection string to confirm echo > .env command works during deploy Action

### CODE REVIEW

File: server/.env.server
Comment (thanhgly) [inline server/.env.server:1]: Is it safe to push this info?

Comment (gbmyt) [reply server/.env.server:1]: Definitely not, thanks for bringing that to my attention-- must have accidentally committed. Removing now

**Review by thanhgly — COMMENTED** [review]
I reviewed the PR.
You are merging the .env.server file to the main branch. Is this on purpose? If it is, it's good to go.

**Comment by gbmyt** [comment]
@thanhgly Mistakenly committed that-- forgot the .gitignore doesn't check for .server/.client postfixes. Can you confirm the file was removed in my last update to this PR? Definitely do not want to merge either of our dotenvs to main. 

Thanks!

**Review by thanhgly — APPROVED** [review]
@gbmyt 
Reviewed.
Merging is blocked due to conflicts. Just resolve the conflicts and it's safe to merge.

**Pull request merged**